.editorconfig

```diff
@@ -1,19 +1,285 @@
-###############################
-# Core EditorConfig Options   #
-###############################
-
-# dotnet-format requires version 3.1.37601
-# dotnet tool update -g dotnet-format
-# remember to have: git config --global core.autocrlf false  #(which is usually default)
+# EditorConfig is awesome: https://EditorConfig.org
 
+# top-most EditorConfig file
 root = true
 
-# Every file
-
+# Don't use tabs for indentation.
 [*]
+indent_style = space
+# (Please don't specify an indent_size here; that has too many unintended consequences.)
+spelling_exclusion_path = SpellingExclusions.dic
+
+# Code files
+[*.{cs,csx,vb,vbx}]
+indent_size = 4
 insert_final_newline = true
 trim_trailing_whitespace = true
 charset = utf-8
+
+# XML project files
+[*.{csproj,vbproj,vcxproj,vcxproj.filters,proj,projitems,shproj}]
+indent_size = 2
+
+# XML config files
+[*.{props,targets,ruleset,config,nuspec,resx,vsixmanifest,vsct}]
+indent_size = 2
+
+# JSON files
+[*.json]
+indent_size = 2
+
+# Powershell files
+[*.ps1]
+indent_size = 2
+
+# Shell script files
+[*.sh]
 end_of_line = lf
+indent_size = 2
+
+# Dotnet code style settings:
+[*.{cs,vb}]
+
+# Sort using and Import directives with System.* appearing first
+dotnet_sort_system_directives_first = true
```

Comment:
```suggestion
dotnet_sort_system_directives_first = false
```

Reply:
Default value is true (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/dotnet-formatting-options)

Reply:
ya but this is annoying, should be alphabetical.

Reply:
Why? I think that if this is the case by default, there is a reason.

Reply:
> ya but this is annoying, should be alphabetical.

The same argument as my `var` :)